config/helpers.php

```diff
@@ -1,154 +1,136 @@
 <?php
 
 use Kirby\Cms\App;
+use Kirby\Cms\Helpers;
 use Kirby\Cms\Html;
-use Kirby\Cms\Response;
 use Kirby\Cms\Url;
-use Kirby\Exception\InvalidArgumentException;
 use Kirby\Filesystem\Asset;
 use Kirby\Filesystem\F;
+use Kirby\Http\Response;
 use Kirby\Http\Router;
 use Kirby\Toolkit\Date;
-use Kirby\Toolkit\Escape;
 use Kirby\Toolkit\I18n;
 use Kirby\Toolkit\Str;
 use Kirby\Toolkit\V;
 
-/**
- * Helper to create an asset object
- *
- * @param string $path
- * @return \Kirby\Filesystem\Asset
- */
-function asset(string $path)
-{
-    return new Asset($path);
-}
-
-/**
- * Generates a list of HTML attributes
- *
- * @param array|null $attr A list of attributes as key/value array
- * @param string|null $before An optional string that will be prepended if the result is not empty
- * @param string|null $after An optional string that will be appended if the result is not empty
- * @return string|null
- */
-function attr(?array $attr = null, ?string $before = null, ?string $after = null): ?string
-{
-    if ($attrs = Html::attr($attr)) {
-        return $before . $attrs . $after;
-    }
-
-    return null;
-}
-
-/**
- * Returns the result of a collection by name
- *
- * @param string $name
- * @return \Kirby\Cms\Collection|null
- */
-function collection(string $name)
-{
-    return App::instance()->collection($name);
-}
-
-/**
- * Checks / returns a CSRF token
- *
- * @param string|null $check Pass a token here to compare it to the one in the session
- * @return string|bool Either the token or a boolean check result
- */
-function csrf(?string $check = null)
-{
-    $session = App::instance()->session();
-
-    // no arguments, generate/return a token
-    // (check explicitly if there have been no arguments at all;
-    // checking for null introduces a security issue because null could come
-    // from user input or bugs in the calling code!)
-    if (func_num_args() === 0) {
-        $token = $session->get('kirby.csrf');
-
-        if (is_string($token) !== true) {
-            $token = bin2hex(random_bytes(32));
-            $session->set('kirby.csrf', $token);
-        }
-
-        return $token;
-    }
-
-    // argument has been passed, check the token
-    if (
-        is_string($check) === true &&
-        is_string($session->get('kirby.csrf')) === true
-    ) {
-        return hash_equals($session->get('kirby.csrf'), $check) === true;
+// @codeCoverageIgnoreStart
+if (
+    defined('KIRBY_HELPER_ASSET') === false ||
+    constant('KIRBY_HELPER_ASSET') !== false
```

Comment:
What do you think about moving these repeats into `Helpers` class to avoid repeats and improve readability? Something like that:

````php
if (Helpers::isHelperDefined('asset') === false) {
  // codes
}
````

### Helpers::isHelperDefined()

````php
function isHelperDefined(string $name) : bool {
  $name = 'KIRBY_HELPER_' . strtoupper($name);
  return defined($name) === true || constant($name) === false
}
````

Reply:
Great idea! 👍

Reply:
I didn't do it in the first place because I think it makes it much harder for people who face conflicts with helpers what to do. Whereas when they would see the explicit if check for the constant, it would be much more self- explanatory.

Reply:
Hm, that's a good point. It's certainly more explicit and requires less digging. On the other hand I feel it's really much cleaner with the helper method. I'd be fine with either solution.